win/pkgs/vsbuildtools.ps1

```diff
@@ -0,0 +1,28 @@
+#Requires -RunAsAdministrator
+
+$ErrorActionPreference="Stop"
+
+$vs_where = "${Env:ProgramFiles(x86)}/Microsoft Visual Studio/Installer/vswhere.exe"
+if (Test-Path $vs_where)
+{
+    $vs_home = & $vs_where -latest -products * -property installationPath
+    if ($vs_home)
+    {
+        Write-Host "Found existing VS installtaion under `"$vs_home`". Skip installation."
+        exit 0
+    }
+}
+
+Invoke-WebRequest -Uri https://aka.ms/vs/16/release/vs_buildtools.exe -OutFile "${Env:SCRATCH}/vs_buildtools.exe"
+
+$install_cmd = "`"${Env:SCRATCH}/vs_buildtools.exe`" --quiet --wait --norestart --nocache " `
+    + "--add Microsoft.VisualStudio.Workload.VCTools;includeRecommended "                   `
+    + "--add Microsoft.VisualStudio.Component.Windows10SDK.16299 "                          `
+    + "--installPath C:\BuildTools || exit /b %ERRORLEVEL%"
+cmd /c $install_cmd
+
+if ($LASTEXITCODE -ne 0 -or $LASTEXITCODE -ne 3003)
```

Comment:
1. `$?` instead of `$LASTEXITCODE`
2. Store it in var before use. It is very easy to accidentally change "last".

Reply:
`2.` is mostly for multiple uses or `$?` not tightly following is main command.

Reply:
No, I think we should use $LastExitCode here.

Here're the differences from the [official doc](https://docs.microsoft.com/en-us/previous-versions//dd347675(v=technet.10)?redirectedfrom=MSDN):
- $LastExitCode
   Contains the exit code of the last Windows-based program that was run. 
- $?  
   Contains the execution status of the last operation. It contains TRUE if the last operation succeeded and FALSE if it failed.

From the vsbuildtools installation links I referenced, we should check the error code here.